scripts/generate_embeddings_openstack.py

```diff
@@ -22,28 +22,45 @@
     print_unreachable_docs_warning
 )
 
+# The OpenStack documentation base URL
 OS_DOCS_ROOT_URL = "https://docs.openstack.org"
 
 
-def os_metadata_func(file_path: str) -> Dict:
-    """Populate metadata for an OpenStack documentation page.
+class OSMetadata(object):
 
-    Args:
-        file_path: str: file path in str
-    """
-    docs_url = lambda file_path: (  # noqa: E731
-        OS_DOCS_ROOT_URL
-        + file_path.removeprefix(EMBEDDINGS_ROOT_DIR).removesuffix("txt")
-        + "html"
-    )
-    return file_metadata_func(file_path, docs_url)
+    def __init__(self, docs_dir, base_url):
+        super(OSMetadata, self).__init__()
+        self._base_path = os.path.abspath(docs_dir)
+        if self._base_path.endswith('/'):
+            self._base_path = self._base_path[:-1]
+
+        self.base_url = base_url
+
+    def set_metadata(self, file_path: str) -> Dict:
+        """Populate metadata for an OpenStack documentation page.
+
+        Args:
+            file_path: str: file path in str
+        """
+        docs_url = lambda file_path: (  # noqa: E731
+            self.base_url
+            + file_path.removeprefix(self._base_path).removesuffix("txt")
+            + "html"
+        )
+        return file_metadata_func(file_path, docs_url)
 
 
 if __name__ == "__main__":
 
     start_time = time.time()
 
     parser = get_common_arg_parser()
+    parser.add_argument(
+        "-w",
+        "--workers",
+        type=int,
+        default=None,
```

Comment:
The logic here is, hm, interesting: https://github.com/run-llama/llama_index/blob/c2a323e96423fcbde131789f1552004eca0fe521/llama-index-core/llama_index/core/readers/file/base.py#L677
May be we should just default to `multiprocessing.cpu_count()` ?

Reply:
Yeah I like the suggestion. The code they have in llama_index is good to prevent someone from asking for more workers than # of CPUs cause that would be inefficient.

I will change the patch to have the cpu_count() as the default! Thanks @syedriko for the suggestion